src/main/java/io/appium/java_client/remote/MobileOptions.java

```diff
@@ -0,0 +1,514 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client.remote;
+
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.MutableCapabilities;
+import org.openqa.selenium.ScreenOrientation;
+import org.openqa.selenium.remote.CapabilityType;
+
+import java.net.URL;
+import java.time.Duration;
+import java.util.List;
+
+public class MobileOptions<T extends MobileOptions<T>> extends MutableCapabilities {
+
+    /**
+     * Creates new instance with no preset capabilities.
+     */
+    public MobileOptions() {
+    }
+
+    /**
+     * Creates new instance with provided capabilities capabilities.
+     *
+     * @param source is Capabilities instance to merge into new instance
+     */
+    public MobileOptions(Capabilities source) {
+        merge(source);
+    }
+
+    /**
+     * Set the Platform Name.
+     *
+     * @param platform the name of the platform.
+     * @return this MobileOptions, for chaining.
+     * @see org.openqa.selenium.remote.CapabilityType#PLATFORM_NAME
+     */
+    public T setPlatformName(String platform) {
+        return amend(CapabilityType.PLATFORM_NAME, platform);
+    }
+
+    /**
+     * Get the Platform Name.
+     *
+     * @return String representing PLATFORM_NAME
+     * @see org.openqa.selenium.remote.CapabilityType#PLATFORM_NAME
+     */
+    public String getPlatformName() {
+        return (String) getCapability(CapabilityType.PLATFORM_NAME);
+    }
+
+    /**
+     * Set the url for the location of the App.
+     *
+     * @param url is a String representing the location of the App
```

Comment:
I'd rather explicitly mention this could be either a full path to a file located on the `same` machine where the server is running or an URL to an application archive/package

Reply:
Sounds good, updating to:

`String`-
`Set the absolute local path for the location of the App.` 
`URL`-
`Set the remote http URL for the location of the App.`

Reply:
4ae42ed81cfa222c8067702c53db70d73056d730